### PR TITLE
Added butanol CRV yaml files in respective isomer folders

### PR DESCRIPTION
**Pull request by morganmayer**

### CODE REVIEW

**Comment by kyleniemeyer** [comment]
@bryanwweber I'll wait for your comments, but the story here is that @morganmayer started over from the master branch here, and added the final files (now using branches for different things!)

**Comment by bryanwweber** [comment]
@kyleniemeyer should these validate with the `master` version of PyKED? If so, there's a few changes that need to be made.

**Comment by kyleniemeyer** [comment]
@morganmayer have you validated these files using the most recent version of PyKED?

**Comment by morganmayer** [comment]
I did validate them, although I'm not sure if it's the most recent version of PyKED, but I will check on that tomorrow.

**Comment by kyleniemeyer** [comment]
@morganmayer Thanks—you should be able to `conda update pyked` and then re-run the validation.

**Comment by bryanwweber** [comment]
@morganmayer If they validate with 0.3.0, the only thing that needs to change is the `chemked-version` should be set to the version the files were validated with (same goes for the toluene data). This would actually be a decent set of files to work on an upgrade script for 0.4.0, but that would be a separate project, not related to this PR :smile:

**Comment by morganmayer** [comment]
I checked the version and I had been using version 0.1.5, so I updated it and it is now 0.3.0, so I will go though and revalidate the files with the correct `chemked-version` as 0.3.0.

**Comment by bryanwweber** [comment]
Also, you might want to make sure that your Git configuration is set properly - it looks like your email address used for the commits isn't the same as the one used for GitHub (which is why it doesn't show your avatar on the commit :smile: )

**Comment by morganmayer** [comment]
Ah, okay, I will take a look at my configuration.

File: 2-butanol/Bec_2014_2-b_CRV.yaml [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+    kind: shock tube
+    institution: Department of Mechanical Engineering, Stanford University
+    facility: stainless steel shock tube
+common-properties:
+    ignition-type:  &ign
+        target: OH*
+        type: d/dt max
+datapoints:
+    - temperature:
+        - 15.6 K
```
Comment (bryanwweber) [inline 2-butanol/Bec_2014_2-b_CRV.yaml]: This looks like a typo :smile: Similar for some of the other cases in this file

File: i-butanol/Bec_2014_i-b_CRV.yaml [hunk 2]
```diff
@@ -0,0 +1,695 @@
...
+    kind: shock tube
+    institution: Department of Mechanical Engineering, Stanford University
+    facility: stainless steel shock tube
+common-properties:
+    ignition-type:  &ign
+        target: OH*
+        type: d/dt max
+datapoints:
+    - temperature:
+        - 1.42 K
```
Comment (bryanwweber) [inline i-butanol/Bec_2014_i-b_CRV.yaml]: Another typo :smile_cat:

File: n-butanol/Bec_2014_n-b_CRV.yaml [hunk 3]
```diff
@@ -0,0 +1,948 @@
...
+    kind: shock tube
+    institution: Department of Mechanical Engineering, Stanford University
+    facility: stainless steel shock tube
+common-properties:
+    ignition-type:  &ign
+        target: OH*
+        type: d/dt max
+datapoints:
+    - temperature:
+        - 16.2 K
```
Comment (bryanwweber) [inline n-butanol/Bec_2014_n-b_CRV.yaml]: Here too :smiley:

File: n-butanol/Bec_2014_n-b_CRV.yaml [hunk 4]
```diff
@@ -0,0 +1,948 @@
...
+      ignition-delay:
+        - 504.0 us
+      pressure:
+        - 1043.0 atm
```
Comment (bryanwweber) [inline n-butanol/Bec_2014_n-b_CRV.yaml]: Maybe pressure and temperature got switched?

**Comment by bryanwweber** [comment]
@morganmayer Sorry, I didn't notice a few typos before. I noted a few of them, but can you check for others?

**Comment by morganmayer** [comment]
@bryanwweber Whoops, yes, it looks like I accidentally switched pressure and temperature. I will fix those and check the other files.

**Comment by morganmayer** [comment]
Sorry for the delay, I remade the files Friday, but I was having strange issues with conda and numpy so I've been trying to figure out how to validate them, but I am close now and should have that done soon today.

**Comment by morganmayer** [comment]
The four Bec butanol files should be correct now, and I deleted three previous Bec i-b CRV files from a while ago that I just combined into the one Bec i-b file instead.

**Comment by bryanwweber** [comment]
@morganmayer I think these look good now. Two things to change:

1. @kyleniemeyer pushed to master without making a PR (:cry:) so now there's a conflict in your PR here. Can you fix that up?
2. Can you check the articles and see if they mention anything about uncertainty in their data?

**Comment by kyleniemeyer** [comment]
I fixed the conflict, since I introduced it...

**Comment by bryanwweber** [comment]
@morganmayer I think this is still waiting for the uncertainties, is that correct? I imagine you're in finals right now, so no rush! Just trying to make sure I'm helping

**Comment by morganmayer** [comment]
Yes, I still need to put in the uncertainties. I've been busy studying for finals this week, but that is what I'm planning to do soon after my exams on Thursday. Thanks! And I'll let you know if I have any questions once I start getting it together!

**Comment by kyleniemeyer** [comment]
@morganmayer is this all ready now, or are you still adding details?

**Comment by morganmayer** [comment]
I've added uncertainties in all of the shock tube files and re-validated all of the butanol files. In the Bec CRV files, should I assume the uncertainties are the same as the shock tube experiment? In the paper, the uncertainties are only mentioned in the shock tube experimental section and not in the CRV section.

**Comment by morganmayer** [comment]
Also I did find one file (Bec_phi_1.43_n-b_0.0338_xO2_0.142.yaml) that had the wrong units of bar when it should have been atm.  I fixed this too and also double checked the pressure units on all of the other files, and the rest were fine.  Sorry about that!

**Comment by morganmayer** [comment]
Ah, just realized I'm not quite done with all of the n-butanol files, I'll finish that real quick

**Comment by kyleniemeyer** [comment]
So, the CRV experiments *are* shock tube experiments, just with a different mixing strategy, so I think uncertainties given would apply to those experiments too.

However, it looks like they only give uncertainties in the postshock temperature and pressure; I'm not sure if those values actually apply to the values we care about, because those are after the reflected shock rather than the incident shock.

**Comment by morganmayer** [comment]
@kyleniemeyer So there shouldn't be uncertainty values for any of the Bec files since there weren't any uncertainties given for initial conditions? I think I was thrown off reading "initial post shock temperature..." I'll check the statements in the other papers for the other uncertainties I used.

The Zhu paper said there was a temporal resolution of 10 us, so I took it as an absolute uncertainty for ignition delay, is that right?

**Comment by kyleniemeyer** [comment]
@morganmayer Yeah, I think that's correct for the Bec data—they don't give uncertainty for the reflected shock measurements.

For Zhu, that seems like a reasonable interpretation. @bryanwweber what do you think?

**Comment by bryanwweber** [comment]
Re: the Bec uncertainties - are the experiments conducted after the incident shock? I thought all the CRV experiments were after the reflected shock, and as such, the uncertainties should be included? The way I read this, the uncertainty is given for the reflected shock conditions:

> The velocity of the incident shock at the end wall was then determined by extrapolation, allowing calculation of the initial reflected shock temperature and pressure, by using one-dimensional shock-jump relations and assuming vibrational equilibrium and frozen chemistry, with uncertainties in initial postshock temperature and pressure of less than ±1.5%.

So the way that I read that is the temperature and pressure should have relative uncertainties of 0.015 in the Bec files. I'm not sure why they don't just measure the pressure and report it, but that's just me. The way we (currently) write the ChemKED files, the reflected shock (T5 and P5) conditions are the "initial" conditions (the `temperature` and `pressure` keys), although that's going to change eventually as we add new shock tube experiments.

Re: The Zhu uncertainties - that seems like the most reasonable interpretation

I just noticed that Bec corrected Zhu's data - which data are encoded here, and why?

**Comment by morganmayer** [comment]
@bryanwweber Sounds good, I will keep the 0.015 relative uncertainties for the Bec files. I'll also add the same uncertainties to the Bec CRV data since they're also shock tube experiments, and likewise for the Zhu CRV data.

This is from the Stranic paper: "Therefore, a linear extrapolation of incident shock speed was used to calculate the incident shock speed at the endwall. This method of calculating the initial post-shock conditions results in an approximate uncertainty of 0.5% in temperature and 1% in pressure .... This results in a temporal resolution of the sidewall OH* emission diagnostic of less than 15 us."  I used these uncertainties in the Stranic files. Just double checking, is that correct?

What exactly do you mean when you say that Bec corrected Zhu's data?

**Comment by bryanwweber** [comment]
@morganmayer In the "Experimental Results and Comparison with Mechanism Prediction" section in the first paragraph they state

> Tables I–V provide a full listing of current test conditions and ignition delay times for iso-, sec-, and tert-butanol, as well as a revised data set (implementing the same correction method for the fuel and oxygen concentration used in these experiments) of Zhu et al. [10] for CRV ignition delay times for n-butanol at 20 atm.

**Comment by morganmayer** [comment]
I just started to reread through the Bec paper, and in the last paragraph of the introduction, it says that data for n-butanol was used from a study by Zhu et al, but the data for the other isomers was their own data collection.  Should I delete the Bec n-butanol files or just add in the detail line they are from the Zhu study? I'm not sure about the method they used to correct Zhu's data.

**Comment by kyleniemeyer** [comment]
@morganmayer If the Bec n-butanol data doesn't overlap with that from the Zhu paper, then I think both should be included. They are separate studies, after all.

It still isn't clear to me how the two studies relate... they were both published around the same time.

**Comment by bryanwweber** [comment]
@kyleniemeyer @morganmayer The way that I interpret this is that Zhu ran his study on the Stanford ST and only studied n-butanol. Bec ran her (I think it's a her) study after Zhu, and decided to study the rest of the isomers. In the course of doing her study, Bec developed a better way to measure the mole fractions of reactants in the ST and decided to go back to Zhu's data and correct the mole fractions. 

If you look at Table V in Bec and Table 1 of Zhu, you'll see that for the first CRV point the T5, P5, and tau are the same, but the XO2 is apparently different. I'm actually not sure what Bec's table means, because she lists the XO2 as 0.203 in the table header and then shows XO2 as 0.162 in the table... Might be worth contacting the corresponding author to figure out what's going on.

**Comment by morganmayer** [comment]
@bryanwweber We double checked with D.F. Davidson about the revised data in the Bec IJCK paper, and those CRV compositions are correct in the IJCK paper, so I deleted the Zhu CRV files. The conventional filling compositions from Zhu should still be correct.  I validated the files with pyked 0.4.1, so these files should be ready to submit.

**Comment by bryanwweber** [comment]
Looks good, thanks @morganmayer

**Pull request merged**